fract4dgui/main_window.py

```diff
@@ -1208,6 +1210,17 @@ def redo(self, *args):
         """Redo an operation after undoing it."""
         self.model.redo()
 
+    def paste(self, *args):
+        """Paste (can be used to update colors)."""
+
+        clipboard = Gtk.Clipboard.get(Gdk.SELECTION_CLIPBOARD)
+        text = clipboard.wait_for_text()
```

Comment:
Returns

a newly-allocated UTF-8 string which must be freed with g_free(), or NULL if retrieving the selection data failed. (This could happen for various reasons, in particular if the clipboard was empty or if the contents of the clipboard could not be converted into text form.). 

Suggest adding a test for `is None`. Else if pasting non-text:

```
gnofract4d/fract4d/gradient.py", line 516, in load_from_url
    if not url.startswith("https://coolors.co/"):
AttributeError: 'NoneType' object has no attribute 'startswith'
```